### PR TITLE
Gradient matching attack example notebooks.

### PR DESCRIPTION
**Pull request by TS-Lee**
# Description

This change adds the example notebooks for the Gradient Matching attack (Witches' Brew). It also makes minor pydoc change for the Gradient Matching attack.

Fixes # (issue)

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS: Ubuntu 18.04.6 LTS
- Python version: 3.8.12
- ART version or commit number: 917a19755a2612cbbcbba85e9d5a218637f1f2d0
- TensorFlow / Keras / PyTorch / MXNet version: Tensorflow==2.4.1, Keras==2.7.0, PyTorch==1.10.2

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by TS-Lee** [comment]
Hi @beat-buesser , do you think this can be merged?
